### PR TITLE
doc: add a basic client configuration document

### PR DESCRIPTION
**Pull request by jtlayton**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/start/client.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    ceph config generate-minimal-conf
+
+This will generate a minimal config file that is mostly just for telling
+the client how to reach the Ceph Monitors. The file should typically be
+installed in `/etc/ceph/ceph.conf`.
+
+Keyring Setup
+=============
+Most Ceph clusters are run with encryption enabled, and the client will
```
Comment (ukernel) [inline doc/start/client.rst]: encryption is not supported so far. I think you mean authentication

Comment (jtlayton) [reply doc/start/client.rst]: Yes, thanks. Will fix.

**Comment by jtlayton** [comment]
jenkins render docs

**Comment by jtlayton** [comment]
jenkins render docs

**Comment by jtlayton** [comment]
Added missing link into the index.

**Comment by jtlayton** [comment]
jenkins render docs

File: doc/start/quick-client.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+for cluster interaction.
+
+Note that most client machines only need the `ceph-common` package and
+its dependencies installed.
+
+Config File Setup
+=================
+Client machines can generally get away with a smaller config file than
+a full-fledged cluster member. To generate a minimal config file, log
+into a machine that is already a cluster member and run::
```
Comment (rishabh-d-dave) [inline doc/start/quick-client.rst]: I don't know what "is already cluster member" means here. Can a machine be part of cluster without having client configured on it or having a Ceph daemon deployed on it?

**Comment by jtlayton** [comment]
jenkins render docs

File: doc/start/quick-client.rst (hunk 2 above)
Comment (jtlayton) [reply doc/start/quick-client.rst]: Good point. I updated the document and hopefully clarified that a bit.

**Comment by jtlayton** [comment]
jenkins render docs

File: doc/start/quick-client.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Note that most client machines only need the `ceph-common` package and
+its dependencies installed. That will supply the basic `ceph` and `rados`
+commands, as well as other commands like `mount.ceph` and `rbd`.
+
+Config File Setup
+=================
+Client machines can generally get away with a smaller config file than
+a full-fledged cluster member. To generate a minimal config file, log
+into a host that is already configured as a client or running a cluster
```
[the remark this answers is not in the document]
Comment (rishabh-d-dave) [reply doc/start/quick-client.rst]: Perhaps, mention `ssh {user}@{mon-host} "sudo ceph config generate-minimal-conf" | sudo tee /etc/ceph/ceph.conf"` instead?

File: doc/start/quick-client.rst [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+daemon, and then run::
+
+    ceph config generate-minimal-conf
+
+This will generate a minimal config file that will tell the client how
+reach the Ceph Monitors. The contents of this file should typically be
```
Comment (rishabh-d-dave) [inline doc/start/quick-client.rst]: s/`how\nreach`/`how to\n reach`

File: doc/start/quick-client.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (jtlayton) [reply doc/start/quick-client.rst]: That's why it says "that is already configured as a client or running a cluster daemon. Not everyone will have ssh access to the monitor, so I'll leave that sort of "recipe" as an exercise for the reader.

File: doc/start/quick-client.rst (hunk 4 above)
Comment (jtlayton) [reply doc/start/quick-client.rst]: Fixed, thanks.

**Comment by varshar16** [comment]
jenkins render docs

File: doc/start/quick-client.rst [hunk 5]
```diff
@@ -0,0 +1,35 @@
+=======================
+Basic Ceph Client Setup
+=======================
+Client machines need some basic configuration in order to interact with
+a cluster. This document describes how to configure a client machine
+for cluster interaction.
+
+Note that most client machines only need the `ceph-common` package and
```
Comment (varshar16) [inline doc/start/quick-client.rst]: Add this to note section.
```suggestion
.. note:: Most of the client machines only need the `ceph-common` package and
```

File: doc/start/quick-client.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (rishabh-d-dave) [reply doc/start/quick-client.rst]: Ok. Makes sense.

File: doc/start/quick-client.rst (hunk 5 above)
Comment (jtlayton) [reply doc/start/quick-client.rst]: Nice! Done.

**Comment by jtlayton** [comment]
jenkins render docs

**Comment by jtlayton** [comment]
jenkins render docs

**Comment by jtlayton** [comment]
jenkins render docs

**Comment by jtlayton** [comment]
I've had this PR sitting around for a while now, but @joscollin asked me this morning about some internal RH docs. I wonder if we ought to also consider adding a "fs-client-setup" command to cephadm, so that you could use it to prep kclients for action. I'll start a thread on the mailing list.

**Comment by jtlayton** [comment]
jenkins render docs

**Comment by jtlayton** [comment]
jenkins render docs

**Comment by jtlayton** [comment]
Same document, but moved to a new location since the old one is going away.

File: doc/cephadm/client-setup.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+installed in `/etc/ceph/ceph.conf`.
+
+Keyring Setup
+=============
+Most Ceph clusters are run with authentication enabled, and the client will
+need keys in order to communicate with cluster machines. To generate a
+keyring file with credentials for `client.fs`, log into an extant cluster
+member and run::
+
+    ceph auth get-or-create client.fs
```
Comment (rishabh-d-dave) [inline doc/cephadm/client-setup.rst]: nit: Using a client name more generic than `client.fs` would be nicer IMO.

**Pull request merged**